File: api/src/main/java/dev/obscuria/fragmentum/registry/DeferredMobEffect.java
package dev.obscuria.fragmentum.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.effect.MobEffect;

import java.util.function.Supplier;

@SuppressWarnings("unused")
public class DeferredMobEffect<T extends MobEffect> extends Deferred<MobEffect, T>
{
    public DeferredMobEffect(Supplier<Holder<MobEffect>> supplier)
    {
        super(supplier);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/registry/DeferredMobEffect.kt
package dev.obscuria.fragmentum.api.registry

import net.minecraft.core.Holder
import net.minecraft.world.effect.MobEffect
import java.util.function.Supplier

class DeferredMobEffect<T : MobEffect>(supplier: Supplier<Holder<MobEffect>>)
    : Deferred<MobEffect, T>(supplier)